### ISSUE TITLE
Cannot send files to/from Firefox browser

### ISSUE THREAD

**Issue by wtquin** [RobinLinus/snapdrop#469]
System info:
- M1 Macbook Air
- Firefox 102.0.1 arm64 (latest stable)
- using snapdrop.net

Whenever I select a file to transfer the snapdrop page freezes (animation wise) and becomes unresponsive. Same thing happens when I use another device to send a file *to* the macbook. Console outputs the following:
```
Uncaught TypeError: this.peers[message.to].sendFiles is not a function
    _onFilesSelected https://snapdrop.net/scripts/network.js:400
    PeersManager https://snapdrop.net/scripts/network.js:369
    fire https://snapdrop.net/scripts/network.js:509
    _onFilesSelected https://snapdrop.net/scripts/ui.js:146
    _bindListeners https://snapdrop.net/scripts/ui.js:111
    _bindListeners https://snapdrop.net/scripts/ui.js:111
    PeerUI https://snapdrop.net/scripts/ui.js:95
    _onPeerJoined https://snapdrop.net/scripts/ui.js:28
    _onPeers https://snapdrop.net/scripts/ui.js:35
    _onPeers https://snapdrop.net/scripts/ui.js:35
    PeersUI https://snapdrop.net/scripts/ui.js:21
    fire https://snapdrop.net/scripts/network.js:509
    _onMessage https://snapdrop.net/scripts/network.js:30
    onmessage https://snapdrop.net/scripts/network.js:19
    _connect https://snapdrop.net/scripts/network.js:19
    ServerConnection https://snapdrop.net/scripts/network.js:7
    Snapdrop https://snapdrop.net/scripts/ui.js:523
    <anonymous> https://snapdrop.net/scripts/ui.js:538
network.js:400:32
    _onFilesSelected https://snapdrop.net/scripts/network.js:400
    PeersManager https://snapdrop.net/scripts/network.js:369
    fire https://snapdrop.net/scripts/network.js:509
    _onFilesSelected https://snapdrop.net/scripts/ui.js:146
    _bindListeners https://snapdrop.net/scripts/ui.js:111
    (Async: EventListener.handleEvent)
    _bindListeners https://snapdrop.net/scripts/ui.js:111
    PeerUI https://snapdrop.net/scripts/ui.js:95
    _onPeerJoined https://snapdrop.net/scripts/ui.js:28
    _onPeers https://snapdrop.net/scripts/ui.js:35
    forEach self-hosted:164
    _onPeers https://snapdrop.net/scripts/ui.js:35
    PeersUI https://snapdrop.net/scripts/ui.js:21
    fire https://snapdrop.net/scripts/network.js:509
    _onMessage https://snapdrop.net/scripts/network.js:30
    onmessage https://snapdrop.net/scripts/network.js:19
    (Async: EventHandlerNonNull)
    _connect https://snapdrop.net/scripts/network.js:19
    ServerConnection https://snapdrop.net/scripts/network.js:7
    Snapdrop https://snapdrop.net/scripts/ui.js:523
    <anonymous> https://snapdrop.net/scripts/ui.js:538
```

If any other device is visible from the snapdrop website, then the page will freeze and become unresponsive, even without sending files, after a few seconds. However, nothing is logged in console when this freeze happens.

**Comment by gokul153** [comment]
i do came across the same thing not working in firefox

**Comment by aha999** [comment]
Same issue here, files aren't being sent/recieved.
Specs: Thinkpad T480, EndeavourOS Linux, KDE Plasma, Firefox & Brave browsers same issue.

**Comment by Quinten0508** [comment]
> By the way, I suggest renaming the title from Firefox browser to Firefox mobile.

No, this happens on the desktop version of Firefox too. **Both** aarch64 and amd64 are affected, the issue seems to be completely platform-agnostic even:
- Windows 11 22H2 Beta 22622.586 (amd64 intel)
- MacOS Ventura beta (aarch64 apple silicon)
- Android 13 (aarch64 qualcomm snapdragon)
- Specs: Thinkpad T480, EndeavourOS Linux, KDE Plasma, Firefox & Brave browsers same issue.  
    _from @/aha999_

**Comment by Quinten0508** [comment]
Just did some extra testing on all the devices I have available:

### Works as expected:
- amd64 - Windows 11: Firefox 104.0.1
- amd64 -Windows 11: Ungoogled Chromium 103
- aarch64 - Android 13: Firefox Nightly 106.0a1
- aarch64 - Android 13: Chromium 104.0
- aarch64 - macOS Ventura 13.0: Ungoogled Chromium 104

It is possible that a webRTC-related bug was patched in one of the later versions of Firefox. Might be worth retrying with the latest Firefox and/or Firefox nightly on your Android device @tbertels 


### Does not work as expected:
- aarch64 - macOS Ventura 13.0: Firefox 104.0.1

Sending a file from chromium to the problematic firefox/macos combo gives me the following console output on chromium: https://hastebin.nl/QFlpH6N

The problematic firefox instance ("receiver" of the file) does not show any additional output in console or network tab. The page does stay responsive to user input, such as clicking on other users, which opens Finder, or the dark/light mode toggle. 
This same instance also still receives webRTC pings, just like all other clients on the network:
```
WS: 
Object { type: "ping" }
​
type: "ping"
​
<prototype>: Object { … }
network.js:27:17
```

It might be worth trying firefox on an intel-based macbook to see if this problem is also present on x64 based machines (as opposed to arm-based w/ apple silicon).

Sending a file _from_ this problematic firefox instance to any other (working) instance results in the following console error in the firefox browser: https://hastebin.nl/4ukuFWU. Same error message as in the original post. Nothing shows up on the `console` tab of the receiving browser, and the `network` tab of both browsers remain empty (excluding already present traffic and pings).

**Comment by SuperKenVery** [comment]
I'm still having issues using snapdrop in firefox...

No errors in console though

AsahiLinux, aarch64, firefox114.0.2

**Comment by SukkaW** [comment]
I am also facing this issue w/ Chrome 117 on macOS Sonoma when sending files to my Pixel 7 w/ Chrome 117. The console on the macOS logs exactly the same error.

**Comment by zhuanghaotao** [comment]
这是来自ZHNG-Haotiu邮箱自动回复。
